src/DeviceStatusPage/DeviceInfoCard.tsx

```diff
@@ -76,51 +77,60 @@ const DeviceInfoCard = ({
   onEnableCredentialsClick,
   onWipeCredentialsClick,
   onDeleteDeviceClick,
-}: DeviceInfoCardProps): React.ReactElement => (
-  <FullHeightCard xs={12} md={6} className="mb-4">
-    <Card.Header as="h5">Device Info</Card.Header>
-    <Card.Body className="d-flex flex-column">
-      <h6>Device ID</h6>
-      <p className="font-monospace">{device.id}</p>
-      <h6>Device name</h6>
-      <p>{device.hasNameAlias ? device.name : 'No name alias set'}</p>
-      <h6>Status</h6>
-      <p>
-        <DeviceStatus status={device.deviceStatus} />
-      </p>
-      <h6>Credentials inhibited</h6>
-      <p>{device.hasCredentialsInhibited ? 'True' : 'False'}</p>
-      <div className="mt-auto d-flex flex-column flex-md-row flex-wrap gap-2">
-        {device.hasCredentialsInhibited ? (
+}: DeviceInfoCardProps): React.ReactElement => {
+  const astarte = useAstarte();
+  return (
+    <FullHeightCard xs={12} md={6} className="mb-4">
+      <Card.Header as="h5">Device Info</Card.Header>
+      <Card.Body className="d-flex flex-column">
+        <h6>Device ID</h6>
+        <p className="font-monospace">{device.id}</p>
+        <h6>Device name</h6>
+        <p>{device.hasNameAlias ? device.name : 'No name alias set'}</p>
+        <h6>Status</h6>
+        <p>
+          <DeviceStatus status={device.deviceStatus} />
+        </p>
+        <h6>Credentials inhibited</h6>
+        <p>{device.hasCredentialsInhibited ? 'True' : 'False'}</p>
+        <div className="mt-auto d-flex flex-column flex-md-row flex-wrap gap-2">
+          {device.hasCredentialsInhibited ? (
+            <Button
+              variant="success text-white"
+              onClick={onEnableCredentialsClick}
+              disabled={device.deletionInProgress}
+            >
+              Enable credentials request
+            </Button>
+          ) : (
+            <Button
+              variant="danger"
+              onClick={onInhibitCredentialsClick}
+              disabled={device.deletionInProgress}
+            >
+              Inhibit credentials
+            </Button>
+          )}
           <Button
-            variant="success text-white"
-            onClick={onEnableCredentialsClick}
+            variant="danger"
+            onClick={onWipeCredentialsClick}
+            hidden={!astarte.token?.can('realmManagement', 'DELETE', `/devices/${device.id}`)}
```

Comment:
The token cannot wipe the credential secret when:

```suggestion
            hidden={!astarte.token?.can('pairing', 'DELETE', `/agent/devices/${device.id}`)}
```